### PR TITLE
Don't delete all lessons and roles when deleting a user :)

### PR DESCRIPTION
**Pull request by gitkuba**

### CODE REVIEW

File: app/Console/Commands/UserPurge.php [hunk 1]
```diff
@@ -62,22 +62,21 @@ public function handle()
...
-
 		$user->orders()->delete();
 		$user->coupons()->delete();
 		$user->profile()->delete();
 		$user->personalData()->delete();
 		$user->billing()->delete();
 		$user->settings()->delete();
 		$user->userAddress()->delete();
-		$user->roles()->delete();
+		$user->roles()->detach();
```
Comment (bognix) [inline app/Console/Commands/UserPurge.php]: what's the guideline here?

On which relations we should use detach and on which delete?

File: app/Console/Commands/UserPurge.php [hunk 2]
```diff
@@ -62,22 +62,21 @@ public function handle()
 	}
 
 	public function purge($user) {
-
 		$user->orders()->delete();
```
Comment (kvas-damian) [inline app/Console/Commands/UserPurge.php]: Now I'm not sure if we should be deleteting orders data. Even if there is no user we should have history of all orders. What do you think?

Comment (bognix) [reply app/Console/Commands/UserPurge.php]: I agree, given we generate invoices on our side.

File: app/Console/Commands/UserPurge.php (hunk 1 above)
Comment (gitkuba) [reply app/Console/Commands/UserPurge.php]: On many to many or morph type relations we should use `detach()`, every other `delete()`. Not 100% sure though.

File: app/Console/Commands/UserPurge.php (hunk 2 above)
Comment (gitkuba) [reply app/Console/Commands/UserPurge.php]: agree

**Comment by kvas-damian** [comment]
BTW, didn't we agree to move this method to User model hook?

**Pull request merged**